Fix/spatial mixins

Marius1311 opened the pull request:
This implements a batched mode for `correlate` and `impute`, offering a temporary fix for https://github.com/theislab/moscot/issues/729.

MUCDK commented on src/moscot/problems/space/_mixins.py:
```diff
@@ -414,10 +415,13 @@ def correlate(  # type: ignore[misc]
         device
             Device where to transfer the solutions, see :meth:`~moscot.base.output.BaseSolverOutput.to`.
+        groupby
+            Optional `obs` field in `AnnData` to compute correlations over categorical groups.
```
I know this was there before but can we do ` `:attr:`~anndata.AnnData.obs ` `

MUCDK replied:
:attr:`~anndata.AnnData.obs

Marius1311 replied:
great, done!

MUCDK reviewed (changes requested):
Other than the minor doc thing everything looks good to me, thanks Marius!

and in the test maybe once test for this.

Marius1311 commented:
> Other than the minor doc thing everything looks good to me, thanks Marius!
> 
> and in the test maybe once test for this.

Thanks for your feedback @MUCDK, I added that minor change to the docs, extended tests to cover this case, spotted a corner case and fixed it. Everything should be fine now.
